From 1adeed7f3f3f3e5e6732ba6f6d6d6df226f1f345 Mon Sep 17 00:00:00 2001
From: Arnaud Ferraris <aferraris@debian.org>
Date: Sat, 11 Feb 2023 23:52:28 +0100
Subject: [PATCH] service: prevent display manager from starting while running

We can end up in situations where droid-juicer is running and the
display manager starts anyway, so the user can login and start working.
However the device will reboot as soon as droid-juicer finishes,
probably disappointing the user. Let's ensure this doesn't happen by
ensuring the display manager won't start while droid-juicer runs.

Origin: Upstream, commit def1459f61452730f0ee481196ca279100d5d03b
---
 droid-juicer.service | 1 +
 1 file changed, 1 insertion(+)

diff --git a/droid-juicer.service b/droid-juicer.service
index 2820891..714f166 100644
--- a/droid-juicer.service
+++ b/droid-juicer.service
@@ -2,10 +2,12 @@
 Description=Android vendor firmware extraction
 Wants=sysinit.target make-dynpart-mappings@super.service
 After=-.mount tmp.mount boot.mount make-dynpart-mappings@super.service
+Before=plymouth-quit.service display-manager.service
 ConditionPathExists=!/var/lib/droid-juicer/status.json
 
 [Service]
 Type=oneshot
+ExecStartPre=-/usr/bin/plymouth display-message --text="Extracting binary firmware..."
 ExecStart=/usr/bin/droid-juicer
 ExecStartPost=/usr/sbin/reboot
 
--
2.39.1
